Include specifics of how reproducibility practices were implemented (#48)
UBC-MDS/teaching_responsibility: manuscript/teaching-reproducibility.Rmd

including undergraduate students taking their first data science course
and graduate students looking to specialize in data science.
We teach the plurality of our courses in the Master of Data Science program,

> which consists of 24 one-credit courses over an eight month period

Author: Added this here and removed the equivalent sentence in the reproducibility section.

culminating in a six-credit-capstone course.
As a pre-requisite for this program

> students are generally required to have taken one course each

Reviewer: On the line below it says stat and prob, but it should be stat or prob

Reviewer: Change lines below from:

```
Typically,
students also have a few years of working experience (or equivalent),
but our cohorts includes both recent graduates from Bachelor's degrees
as well as learners from senior company positions
looking to bend their career trajectories towards data science.
```

to:

```
Most of these graduate students have several years of professional work experience (or equivalent),
however our cohorts also include both recent graduates from Bachelor's degrees
as well as learners from senior company positions
looking to bend their career trajectories towards data science.
```

grade while completing assignments focused on other data science topics.
This latter strategy parallels how they will employ reproducibility 
habits later in their carers
(a detailed example follows in the next section).
The tools we chose to teach with are data science reproducibility software
that students are likely to employ in workflows in their future work.
This sustained practice not only enforces students' habits,
but also increases their proficiency using authentic reproducibility software
and when they into problems, it is in an environment

> where they can easily reach out for help without feeling intimidated to ask.

Reviewer: Why was this removed? I like it. If there is good reason to remove it however, please let me know and we can do so.

Reviewer: Ohhhhh, now I see, this was just moved below! You can ignore the above comment.

> (a detailed example follows in the next section),

Reviewer: ```suggestion
(a detailed example follows in the next section).
```

> during the first few weeks of class.

Reviewer: This should be it's own sentence. And it needs to be edited. I find it hard to parse and digest. Or we could remove it?

> To engage students at different levels of skill
> and interest in reproducibility,
> we have more difficult exercises
> in the courses specifically teaching these topics
> (including demanding optional exercises for bonus points).
> Students less experienced and interested in these topics
> have shared with us that they find the worksheets
> really helpful in preparing them
> to successfully complete the lab questions.

Reviewer: ```suggestion
We have observed a wide-range of interest, experience 
and skill-level in our students when teaching reproducibility.
Feedback from instructor and student-led surveys have indicated that
the guided electronic worksheets with automated feedback
have increased students perceived learning.
We hypothesize this is particularly true for students' who
have less interest, experience or skills in the area of reproducibility.
To challenge students who have more interest, experience or skills in the area of reproducibility,
we routinely include demanding optional questions in our homework assignments 
that allow these students to engage more deeply with the material for a small amount of bonus points.
We hope these strategies help to engage students across the wide range of interest, experience 
and skills in the area of reproducibility that we see in our classrooms.
```

The decision to give students plenty of opportunities to practice reproducible habits
by adopting version control for student homework submission
was implemented across the Master of Data Science program from year one.
To make this feasible,
postdoctoral teaching and learning fellows were hired
to help faculty implement this in their courses
even if those faculty themselves were not familiar with version control.
The teaching team model in the Master of Data Science program
greatly facilitates us doing program-wide changes like this.
This stems from our shared vision for the program
(which was drafted collaboratively by the team),

> our once-per term academic retreats where we reflect on how things went,

Reviewer: ```suggestion
our once-per term academic retreats where we reflect on how things went in each course,
```

including [GitHub Classroom](https://classroom.github.com/) 
and [Classy](https://github.com/ubccpsc/classy).

> To incentivize doing students to submit their assignment by using Git 

Reviewer: ```suggestion
To incentivize students to submit their assignment using Git 
```
